cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -115,8 +115,8 @@ type ScaleUpFailure struct {
 type ClusterStateRegistry struct {
 	sync.Mutex
 	config                             ClusterStateRegistryConfig
-	scaleUpRequests                    map[string]*ScaleUpRequest // nodeGroupName -> ScaleUpRequest
-	scaleDownRequests                  []*ScaleDownRequest
+	scaleUpRequests                    map[string]*ScaleUpRequest   // nodeGroupName -> ScaleUpRequest
+	scaleDownRequests                  map[string]*ScaleDownRequest // nodeGroupName -> ScaleDownRequest
```

Comment:
I have changed `scaleDownRequests` to map to make it more convenient for use.
Earlier:
```go
csr.scaleDownRequests = append(csr.scaleDownRequests, scaleDownReq)
```
Now:
```go
csr.scaleDownRequests[nodeGroup.Id()] = scaleDownReq
```